modules_namespace.go

```diff
@@ -0,0 +1,182 @@
+	idx int
+}
+
+func (no *namespaceObject) iterateStringKeys() iterNextFunc {
+	return (&namespacePropIter{
+		no: no,
+	}).next
+}
+
+func (no *namespaceObject) iterateKeys() iterNextFunc {
+	return no.iterateStringKeys()
+}
+
+func (i *namespacePropIter) next() (propIterItem, iterNextFunc) {
+	for i.idx < len(i.no.exportsNames) {
+		name := i.no.exportsNames[i.idx]
+		i.idx++
+		prop := i.no.getOwnPropStr(name)
+		if prop != nil {
+			return propIterItem{name: stringValueFromRaw(name), value: prop}, i.next
+		}
+	}
+	return propIterItem{}, nil
+}
+
+func (no *namespaceObject) getOwnPropStr(name unistring.String) Value {
+	if _, ok := no.exports[name]; !ok {
+		return nil
+	}
+	v, ambiguous := no.m.ResolveExport(name.String())
+	if ambiguous || v == nil {
+		no.val.runtime.throwReferenceError((name))
+	}
+	if v.BindingName == "*namespace*" {
+		return &valueProperty{
+			value:        no.val.runtime.NamespaceObjectFor(v.Module),
+			writable:     true,
+			configurable: false,
+			enumerable:   true,
+		}
+	}
+
+	mi := no.val.runtime.modules[v.Module]
+	b := mi.GetBindingValue(v.BindingName)
+	if b == nil {
+		// TODO figure this out - this is needed as otherwise valueproperty is thought to not have a value
+		// which isn't really correct in a particular test around isFrozen
+		b = _null
+	}
+	return &valueProperty{
+		value:        b,
+		writable:     true,
+		configurable: false,
+		enumerable:   true,
+	}
+}
+
+func (no *namespaceObject) hasOwnPropertyStr(name unistring.String) bool {
+	_, ok := no.exports[name]
+	return ok
+}
+
+func (no *namespaceObject) getStr(name unistring.String, receiver Value) Value {
+	prop := no.getOwnPropStr(name)
+	if prop, ok := prop.(*valueProperty); ok {
+		if receiver == nil {
+			return prop.get(no.val)
+		}
+		return prop.get(receiver)
+	}
+	return prop
+}
+
+func (no *namespaceObject) setOwnStr(name unistring.String, val Value, throw bool) bool {
+	no.val.runtime.typeErrorResult(throw, "Cannot add property %s, object is not extensible", name)
+	return false
+}
+
+func (no *namespaceObject) deleteStr(name unistring.String, throw bool) bool {
+	if _, exists := no.exports[name]; exists {
+		no.val.runtime.typeErrorResult(throw, "Cannot add property %s, object is not extensible", name)
+		return false
+	}
+	return true
+}
+
+func (no *namespaceObject) defineOwnPropertyStr(name unistring.String, desc PropertyDescriptor, throw bool) bool {
+	returnFalse := func() bool {
+		if throw {
+			no.val.runtime.typeErrorResult(throw, "Cannot add property %s, object is not extensible", name)
+		}
+		return false
+	}
+	if !no.hasOwnPropertyStr(name) {
+		return returnFalse()
+	}
+	if desc.Empty() {
+		return true
+	}
+	if desc.Writable == FLAG_FALSE {
+		return returnFalse()
+	}
+	if desc.Configurable == FLAG_TRUE {
+		return returnFalse()
+	}
+	if desc.Enumerable == FLAG_FALSE {
+		return returnFalse()
+	}
+	if desc.Value != nil && desc.Value != no.getOwnPropStr(name) {
+		return returnFalse()
+	}
+	// TODO more checks
```

Comment:
```suggestion
	// TODO more checks
```
Is it expected to not do them now?

Reply:
There weren't any more things to check actually